skbio/alignment/tests/test_tabular_msa.py

```diff
@@ -901,6 +899,876 @@ def test_with_index(self):
+
+        self.assertEqual(new0, a)
+        self.assertEqual(new1, b)
+
+    def test_get_scalar_second_axis(self):
+        a = DNA("AA", metadata={'a': 'foo'}, positional_metadata={'x': [1, 2]})
+        b = DNA("GC", metadata={'b': 'bar'}, positional_metadata={'y': [3, 4]})
+        msa = TabularMSA([a, b], positional_metadata={'z': [5, 6]})
+
+        new0 = self.get(msa, (Ellipsis, 0))
+        new1 = self.get(msa, (Ellipsis, 1))
+
+        self.assertEqual(new0,
+                         Sequence("AG", metadata={'z': 5},
+                                  positional_metadata={'x': [1, np.nan],
+                                                       'y': [np.nan, 3]}))
+        self.assertEqual(new1,
+                         Sequence("AC", metadata={'z': 6},
+                                  positional_metadata={'x': [2, np.nan],
+                                                       'y': [np.nan, 4]}))
+
+    def test_scalar_sliced_first_axis(self):
+        a = DNA("ACGT", metadata={0: 0}, positional_metadata={0: [1, 2, 3, 4]})
+        b = DNA("ACGT", metadata={1: 1}, positional_metadata={1: [1, 2, 3, 4]})
+        c = DNA("ACGT", metadata={2: 2}, positional_metadata={2: [1, 2, 3, 4]})
+
+        msa = TabularMSA([a, b, c], metadata={3: 3},
+                         positional_metadata={3: [1, 2, 3, 4]})
+
+        self.assertEqual(self.get(msa, (1, [1, 3])),
+                         DNA("CT", metadata={1: 1},
+                             positional_metadata={1: [2, 4]}))
+
+    def test_scalar_sliced_second_axis(self):
+        a = DNA("ACGT", metadata={0: 0}, positional_metadata={0: [1, 2, 3, 4]})
+        b = DNA("ACGA", metadata={1: 1}, positional_metadata={1: [1, 2, 3, 4]})
+        c = DNA("ACGT", metadata={2: 2}, positional_metadata={2: [1, 2, 3, 4]})
+
+        msa = TabularMSA([a, b, c], metadata={3: 3},
+                         positional_metadata={3: [1, 2, 3, 4]})
+
+        self.assertEqual(self.get(msa, ([1, 2], 3)),
+                         Sequence("AT", metadata={3: 4},
+                                  positional_metadata={1: [4, np.nan],
+                                                       2: [np.nan, 4]}))
+
+    def test_get_scalar_out_of_bound_first_axis(self):
+        a = DNA("AA", metadata={'a': 'foo'}, positional_metadata={'x': [1, 2]})
+        b = DNA("GC", metadata={'b': 'bar'}, positional_metadata={'y': [3, 4]})
+        msa = TabularMSA([a, b], positional_metadata={'z': [5, 6]})
+
+        with self.assertRaises(IndexError):
+            self.get(msa, 3)
+
+    def test_get_scalar_out_of_bound_second_axis(self):
+        a = DNA("AA", metadata={'a': 'foo'}, positional_metadata={'x': [1, 2]})
+        b = DNA("GC", metadata={'b': 'bar'}, positional_metadata={'y': [3, 4]})
+        msa = TabularMSA([a, b], positional_metadata={'z': [5, 6]})
+
+        with self.assertRaises(IndexError):
+            self.get(msa, (..., 3))
+
+
+class TestGetItem(SharedIndexTests, unittest.TestCase):
```

Comment:
What about having this subclass `TestILoc`? Then `test_uses_iloc_not_loc` can be removed and `__getitem__` will get tested the same as `iloc`.